napari/components/dims.py

```diff
@@ -140,54 +171,106 @@ def _check_dims(cls, values):
-        if len(values['axis_labels']) < ndim:
+        labels_ndim = len(values['axis_labels'])
+        if labels_ndim < ndim:
             # Append new "default" labels to existing ones
-            if values['axis_labels'] == tuple(
-                map(str, range(len(values['axis_labels'])))
-            ):
+            if values['axis_labels'] == tuple(map(str, range(labels_ndim))):
                 values['axis_labels'] = tuple(map(str, range(ndim)))
             else:
                 values['axis_labels'] = (
-                    tuple(map(str, range(ndim - len(values['axis_labels']))))
+                    tuple(map(str, range(ndim - labels_ndim)))
                     + values['axis_labels']
                 )
-        elif len(values['axis_labels']) > ndim:
+        elif labels_ndim > ndim:
             values['axis_labels'] = values['axis_labels'][-ndim:]
 
         return values
 
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        # to be removed if/when deprecating current_step
+        self.events.point_slider.connect(self.events.current_step)
+
     @property
-    def nsteps(self) -> Tuple[int, ...]:
-        """Tuple of int: Number of slider steps for each dimension."""
+    def nsteps(self) -> Tuple[float]:
         return tuple(
-            int((max_val - min_val) / step_size)
-            for min_val, max_val, step_size in self.range
+            # "or 1" ensures degenerate dimension works
+            int((rng.stop - rng.start) / (rng.step or 1))
+            for rng in self.range
         )
 
+    @nsteps.setter
+    def nsteps(self, value):
+        self.range = [
+            (rng.start, rng.stop, (rng.stop - rng.start) / nsteps)
+            for rng, nsteps in zip(self.range, value)
+        ]
+
     @property
-    def point(self) -> Tuple[int, ...]:
-        """Tuple of float: Value of each dimension."""
-        # The point value is computed from the range and current_step
-        point = tuple(
-            min_val + step_size * value
-            for (min_val, max_val, step_size), value in zip(
-                self.range, self.current_step
-            )
+    def current_step(self) -> Tuple[int]:
+        warnings.warn(
+            trans._(
+                'Dims.current_step is deprecated. Use Dims.point_slider instead.'
+            ),
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        return self.point_slider
+
+    @current_step.setter
+    def current_step(self, value: Tuple[int]):
+        warnings.warn(
+            trans._(
+                'Dims.current_step is deprecated. Use Dims.point_slider instead.'
+            ),
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        self.point_slider = value
+
+    @property
+    def point_slider(self):
+        return tuple(
+            int(round((point - rng.start) / (rng.step or 1)))
+            for point, rng in zip(self.point, self.range)
+        )
```

Comment:
Isn't range in world coordinates? ie this is *not* actually the same as *current_step* was, is it?

Reply:
Yes, it is, cause it's divided by the step. This gives you the point in slider coordinates (just like `current_step` did). Or am I misunderstanding the question?

Reply:
Nope, you're not misunderstanding, I am just bad at reading apparently 🤣  Thank you! 😅